### PR TITLE
New options for category search; Bugfix; Search SDK patch release

### PR DESCRIPTION
**Pull request by DzmitryFomchyn**
### Description
This PR
- Cherry picks latest changes from the Search SDK v2.7.0-rc.1
- Adds new category search option - `CategorySearchOptions.attributeSets`
- Releases patch v2.6.3

### Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR (where applicable)
- [ ] I have run tests and automatic checks locally
- [ ] I have run `pitest` check locally and checked that the coverage increased (or didn't change) or decreased insignificantly
- [ ] I have made corresponding changes to the documentation (where applicable)
- [ ] I have grouped commits logically or I promise to squash my commits before merge

### CODE REVIEW

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/ForwardSearchOptions.kt [hunk 1]
```diff
@@ -362,5 +362,8 @@ internal fun ForwardSearchOptions.mapToCore(): CoreSearchOptions = CoreSearchOpt
     route = null,
     sarType = null,
     timeDeviation = null,
-    addonAPI = unsafeParameters?.let { (it as? HashMap) ?: HashMap(it) }
+    addonAPI = unsafeParameters?.let { (it as? HashMap) ?: HashMap(it) },
+    ensureResultsPerCategory = null,
+    // TODO FIXME should we support attribute sets for the forward geocoding
+    attributeSets = null,
```
Comment (mikeringrose) [inline MapboxSearch/sdk/src/main/java/com/mapbox/search/ForwardSearchOptions.kt:368]: This is coming in an upcoming searchbox release.

**Review by mikeringrose — APPROVED** [review]
Looks good. One thing to note is that Category search will always return the basic information even when not specified. For example, this will return basic attributes:
```
curl --silent --location "https://api.mapbox.com/search/searchbox/v1/category/restaurant?access_token=$MAPBOX_ACCESS_TOKEN&language=en&navigation_profile=driving&proximity=11.966371,57.709765&origin=11.966371,57.709765&attribute_sets=visit&limit=1"
```
The Details API doesn't operate this way. I'm not sure if it's worth noting the difference in the comments. For example, not even the name is returned in this request:
```
❯ curl --silent --location "https://api.mapbox.com/search/details/v1/retrieve/dXJuOm1ieHBvaTpiMDcyZmI2Zi1lNDZkLTQ2MDYtYTJmMy04OTk0N2Q4MmVlMzU?attribute_sets=photos&access_token=$MAPBOX_ACCESS_TOKEN" | jq
{
  "geometry": {
    "coordinates": [
      11.965947,
      57.709667
    ],
    "type": "Point"
  },
  "properties": {
    "mapbox_id": "dXJuOm1ieHBvaTpiMDcyZmI2Zi1lNDZkLTQ2MDYtYTJmMy04OTk0N2Q4MmVlMzU",
    "metadata": {
      "photos": [
        {
          "height": 768,
          "url": "https://media-cdn.tripadvisor.com/media/photo-o/16/f1/b5/9f/mat.jpg",
          "width": 1152
        },
        {
          "height": 1152,
          "url": "https://media-cdn.tripadvisor.com/media/photo-o/16/f1/b5/84/mat.jpg",
          "width": 768
        },
        {
          "height": 1152,
          "url": "https://media-cdn.tripadvisor.com/media/photo-o/16/f1/b4/67/mat.jpg",
          "width": 768
        },
        {
          "height": 2420,
          "url": "https://media-cdn.tripadvisor.com/media/photo-o/19/c3/fb/2a/raspberry-passionfruit.jpg",
          "width": 2420
        },
        {
          "height": 768,
          "url": "https://media-cdn.tripadvisor.com/media/photo-o/16/f1/b5/9a/mat.jpg",
          "width": 1129
        }
      ]
    }
  },
  "type": "Feature"
}
```

File: MapboxSearch/sdk/src/main/java/com/mapbox/search/ForwardSearchOptions.kt (hunk 1 above)
Comment (DzmitryFomchyn) [reply MapboxSearch/sdk/src/main/java/com/mapbox/search/ForwardSearchOptions.kt]: Let's leave it as is for now. It's not required in this patch

**Comment by DzmitryFomchyn** [comment]
> Looks good. One thing to note is that Category search will always return the basic information even when not specified. For example, this will return basic attributes:
> 
> ```
> curl --silent --location "https://api.mapbox.com/search/searchbox/v1/category/restaurant?access_token=$MAPBOX_ACCESS_TOKEN&language=en&navigation_profile=driving&proximity=11.966371,57.709765&origin=11.966371,57.709765&attribute_sets=visit&limit=1"
> ```
> 
> The Details API doesn't operate this way. I'm not sure if it's worth noting the difference in the comments. For example, not even the name is returned in this request:
> 
> ```
> ❯ curl --silent --location "https://api.mapbox.com/search/details/v1/retrieve/dXJuOm1ieHBvaTpiMDcyZmI2Zi1lNDZkLTQ2MDYtYTJmMy04OTk0N2Q4MmVlMzU?attribute_sets=photos&access_token=$MAPBOX_ACCESS_TOKEN" | jq
> {
>   "geometry": {
>     "coordinates": [
>       11.965947,
>       57.709667
>     ],
>     "type": "Point"
>   },
>   "properties": {
>     "mapbox_id": "dXJuOm1ieHBvaTpiMDcyZmI2Zi1lNDZkLTQ2MDYtYTJmMy04OTk0N2Q4MmVlMzU",
>     "metadata": {
>       "photos": [
>         {
>           "height": 768,
>           "url": "https://media-cdn.tripadvisor.com/media/photo-o/16/f1/b5/9f/mat.jpg",
>           "width": 1152
>         },
>         {
>           "height": 1152,
>           "url": "https://media-cdn.tripadvisor.com/media/photo-o/16/f1/b5/84/mat.jpg",
>           "width": 768
>         },
>         {
>           "height": 1152,
>           "url": "https://media-cdn.tripadvisor.com/media/photo-o/16/f1/b4/67/mat.jpg",
>           "width": 768
>         },
>         {
>           "height": 2420,
>           "url": "https://media-cdn.tripadvisor.com/media/photo-o/19/c3/fb/2a/raspberry-passionfruit.jpg",
>           "width": 2420
>         },
>         {
>           "height": 768,
>           "url": "https://media-cdn.tripadvisor.com/media/photo-o/16/f1/b5/9a/mat.jpg",
>           "width": 1129
>         }
>       ]
>     }
>   },
>   "type": "Feature"
> }
> ```

Thanks for the note, Mike, I'll remove attribute sets options patching for the category search.

**Pull request merged**